assets/js/modules/adsense/components/widgets/TopEarningContentWidget.js

```diff
@@ -29,15 +29,164 @@ import { compose } from '@wordpress/compose';
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
+import {
+	CORE_USER,
+	KM_ANALYTICS_ADSENSE_TOP_EARNING_CONTENT,
+} from '../../../../googlesitekit/datastore/user/constants';
+import { DATE_RANGE_OFFSET } from '../../datastore/constants';
+import {
+	MetricTileTable,
+	MetricTileTablePlainText,
+} from '../../../../components/KeyMetrics';
+import Link from '../../../../components/Link';
+import { ZeroDataMessage } from '../../../analytics/components/common';
+import { numFmt } from '../../../../util';
 import whenActive from '../../../../util/when-active';
 import ConnectGA4CTATileWidget from '../../../analytics-4/components/widgets/ConnectGA4CTATileWidget';
+import useViewOnly from '../../../../hooks/useViewOnly';
+import { AdSenseLinkCTA } from '../common';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import { MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
 import ConnectAdSenseCTATileWidget from './ConnectAdSenseCTATileWidget';
+const { useSelect, useInViewSelect } = Data;
 
 function TopEarningContentWidget( { Widget } ) {
+	const viewOnlyDashboard = useViewOnly();
+
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+
+	const reportOptions = {
+		...dates,
+		dimensions: [ 'pagePath' ],
+		metrics: [ { name: 'totalAdRevenue' } ],
+		orderby: [
+			{
+				metric: { metricName: 'totalAdRevenue' },
+				desc: true,
+			},
+		],
+		limit: 3,
+	};
+
+	const report = useInViewSelect( ( select ) => {
+		return select( MODULES_ANALYTICS_4 ).getReport( reportOptions );
+	} );
+
+	const error = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			reportOptions,
+		] )
+	);
+
+	const titles = useInViewSelect( ( select ) =>
+		! error
+			? select( MODULES_ANALYTICS_4 ).getPageTitles(
+					report,
+					reportOptions
+			  )
+			: undefined
+	);
+
+	const loading = useSelect(
+		( select ) =>
+			! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+				'getReport',
+				[ reportOptions ]
+			) || titles === undefined
+	);
+
+	const isAdSenseLinked = useSelect( ( select ) => {
+		if ( viewOnlyDashboard && loading ) {
+			return undefined;
+		}
+
+		return select( MODULES_ANALYTICS ).getAdsenseLinked();
```

Comment:
Hi @benbowler, this is using the wrong `getAdsenseLinked()` selector, it should use the `analytics-4` version:

```suggestion
		return select( MODULES_ANALYTICS_4 ).getAdSenseLinked();

```